tests/snuba/api/endpoints/test_organization_events_stats.py

```diff
@@ -572,3 +573,373 @@ def test_invalid_interval(self):
+
+        data = response.data
+
+        assert response.status_code == 200, response.content
+        assert len(data) == 2
+
+        results = data[self.transaction.transaction]
+        assert [attrs for time, attrs in results["data"]] == [
+            [{"count": 3}],
+            [{"count": 0}],
+        ]
+
+        results = data[transaction2.transaction]
+        assert [attrs for time, attrs in results["data"]] == [
+            [{"count": 1}],
+            [{"count": 0}],
+        ]
+
+    def test_top_events_with_query(self):
+        transaction_data = load_data("transaction")
+        transaction_data["start_timestamp"] = iso_format(self.day_ago + timedelta(minutes=2))
+        transaction_data["timestamp"] = iso_format(self.day_ago + timedelta(minutes=6))
+        transaction_data["transaction"] = "/foo_bar/"
+        self.store_event(transaction_data, project_id=self.project.id)
+        with self.feature("organizations:discover-basic"):
+            response = self.client.get(
+                self.url,
+                data={
+                    "start": iso_format(self.day_ago),
+                    "end": iso_format(self.day_ago + timedelta(hours=1, minutes=59)),
+                    "interval": "1h",
+                    "yAxis": "count()",
+                    "orderby": ["-p99()"],
+                    "query": "transaction:/foo_bar/",
+                    "field": ["transaction", "avg(transaction.duration)", "p99()"],
+                    "topEvents": 5,
+                },
+                format="json",
+            )
+
+        data = response.data
+
+        assert response.status_code == 200, response.content
+        assert len(data) == 1
+
+        transaction2_data = data["/foo_bar/"]
+        assert [attrs for time, attrs in transaction2_data["data"]] == [
+            [{"count": 1}],
+            [{"count": 0}],
+        ]
+
+    def test_top_events_with_rpm(self):
+        with self.feature("organizations:discover-basic"):
+            response = self.client.get(
+                self.url,
+                data={
+                    "start": iso_format(self.day_ago),
+                    "end": iso_format(self.day_ago + timedelta(hours=1, minutes=59)),
+                    "interval": "1h",
+                    "yAxis": "rpm()",
+                    "orderby": ["-count()"],
+                    "field": ["message", "user.email", "count()"],
+                    "topEvents": 5,
+                },
+                format="json",
+            )
+
+        data = response.data
+        assert response.status_code == 200, response.content
+        assert len(data) == 5
+
+        for index, event in enumerate(self.events[:5]):
+            message = event.message or event.transaction
+            results = data[
+                ",".join([message, self.event_data[index]["data"]["user"].get("email", "null")])
+            ]
+            assert [{"count": self.event_data[index]["count"] / (3600.0 / 60.0)}] in [
```

Comment:
Will the value always be `count` when getting other yaxis values?

Reply:
Yup, when something is single axis, the values will always be `count`. When its multi axis the results dictionary uses the axis as the key and the values are the same lists of touples but keeps the same format with `count` :+1: